main.c

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright (C) 2019 Ripjar Limited
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * or see see <https://www.gnu.org/licenses/>.
+ */
+
+#include <node_api.h>
+#include <openssl/rsa.h>
+#include <openssl/pem.h>
+#include <openssl/bio.h>
+#include <openssl/x509.h>
+
+#include <string.h>
+
+static napi_value
+x509_cert_pub_key (napi_env env, napi_callback_info info)
+{
+  size_t argc = 1, size;
+  napi_value argv[argc];
+  char *buf;
+  napi_status status = napi_ok;
+
+  BIO *bio;
+  X509 *x509;
+  EVP_PKEY *evp_pubkey;
+  RSA *public_key;
+  char *n_hex, *e_hex;
+  napi_value n_val, e_val;
+  napi_value obj;
+
+  if (napi_get_cb_info (env, info, &argc, argv, NULL, NULL) != napi_ok)
+    {
+      napi_throw_error (env, NULL, "Failed to parse arguments");
+      return NULL;
+    }
+  if (argc < 1)
+    {
+      napi_throw_error (env, NULL, "This function requires one argument");
+      return NULL;
+    }
+  if (napi_get_value_string_utf8 (env, argv[0], NULL, 0, &size) != napi_ok)
+    {
+      napi_throw_error (env, NULL, "Failed to parse string");
+      return NULL;
+    }
+  if ((buf = malloc (++size)) == NULL)
+    {
+      napi_throw_error (env, NULL, "Failed to allocate memory");
+      return NULL;
+    }
+  if (napi_get_value_string_utf8 (env, argv[0], buf,
+				  size, &size) != napi_ok)
+    {
+      napi_throw_error (env, NULL, "Failed to parse string");
```

Comment:
this is the same error message as above, id change it so you know where it wrong

Reply:
It's because it's the same function as above. The one above just checks the size of the string and the second copies the string to a c string